src/test/java/org/mule/tools/apikit/output/MuleConfigGeneratorTest.java

```diff
@@ -110,6 +104,41 @@ private GenerationModel mockGenerationModel(String example) {
     return flowEntry;
   }
 
+  @Test
+  public void blankDocumentWithoutExternalizedGlobalsAndAPIAutodiscovery() {
```

Comment:
rescaffolding a mule-config with api-autodiscovery in global and another with api-autodiscovery in main xml 
rescaffolding with mule-config with a global containing http-listener-config and apikit-config
rescaffolding with http-listener-config in global but apikit-config in main xml and inverse case
please @juanchib and @dcourtil add more test cases